soroban-sdk/src/deploy.rs

```diff
@@ -54,85 +57,76 @@ impl Deployer {
     }
 
     /// Get a deployer that deploys contracts that derive their contract IDs
-    /// from the current contract and the provided salt.
-    pub fn with_current_contract(
-        &self,
-        salt: &impl IntoVal<Env, BytesN<32>>,
-    ) -> DeployerWithCurrentContract {
-        let env = self.env();
-        DeployerWithCurrentContract {
-            env: env.clone(),
-            salt: salt.into_val(env),
+    /// from the current contract.
+    pub fn with_current_contract(&self) -> DeployerWithAddress {
+        DeployerWithAddress {
+            env: self.env.clone(),
+            address: self.env.current_contract_address(),
         }
     }
 
-    #[doc(hidden)]
-    /// Get a deployer for contracts that derive their contract IDs from the
-    /// given contract ID and the provided salt.
-    pub fn with_other_contract(
-        &self,
-        contract_id: &Address,
-        salt: &impl IntoVal<Env, BytesN<32>>,
-    ) -> DeployerWithOtherContract {
-        let env = self.env();
-        DeployerWithOtherContract {
-            env: env.clone(),
-            contract_id: contract_id.clone(),
-            salt: salt.into_val(env),
+    /// Get a deployer that deploys contracts that derive their contract IDs
+    /// from the provided address.
+    ///
+    /// The deployer address must authorize all the deployments.
+    pub fn with_address(&self, address: Address) -> DeployerWithAddress {
+        DeployerWithAddress {
+            env: self.env.clone(),
+            address,
         }
     }
+
+    /// Deploys a new instance of Stellar Asset Contract corresponding to
+    /// provided serialized asset.
+    ///
+    /// `serialized_asset` is the Stellar `Asset` XDR serialized to bytes. Refer
+    /// to `stellar-xdr` create for the exact definition.
+    pub fn deploy_stellar_asset(&self, serialized_asset: impl IntoVal<Env, Bytes>) -> Address {
```

Comment:
This function should be inside a Deployer, and have a corresponding `id` method that can give you the ID of the contract that would be created.